### PR TITLE
[CDAP-15389] Improves plugin reference section to adhere to material spec.

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds `Markdown` component to render markdown content in plugin reference section.
- Introduces custom elements that are best suited for plugin reference section (headings and tables).

_Associated PRs:_
- https://github.com/data-integrations/google-cloud/pull/102
- https://github.com/cdapio/hydrator-plugins/pull/909
- https://github.com/data-integrations/kafka-plugins/pull/60
- https://github.com/data-integrations/database-plugins/pull/34
- https://github.com/data-integrations/salesforce/pull/32

_Note:_
- Need to see if we could use the table component we already have. Right now using material-ui's table component. Since we have an existing JIRA to define table component specs, waiting for that to get in before using it in markdown.

JIRA: https://issues.cask.co/browse/CDAP-15389
Build: https://builds.cask.co/browse/CDAP-UDUT302

### CODE REVIEW

File: cdap-ui/app/cdap/components/Markdown/MarkdownTable.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      );
+  return (
+    <Paper className={classes.root}>
+      <Table>
+        <TableHead>
+          <TableRow className={classes.row}>
+            {flatFilterMap(flatFilterMap(flatFilterMap(children, 'thead'), 'tr'))
+              .flat()
+              .map((head) => (
+                <CustomTableCell>{head} </CustomTableCell>
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/Markdown/MarkdownTable.tsx]: missing key prop?

nit: extra space after `{head}`

File: cdap-ui/app/cdap/components/Markdown/MarkdownTable.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+              ))}
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {flatFilterMap(flatFilterMap(children, 'tbody'), 'tr')
+            .map((tr) => tr.filter((t) => t))
+            .map((row) => {
+              return (
+                <TableRow className={classes.row}>
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/Markdown/MarkdownTable.tsx]: missing key props for these nested mapped components

**Review by elfenheart — APPROVED** [review]
👍 LGTM

**Comment by ajainarayanan** [comment]
Latest green build: https://builds.cask.co/browse/CDAP-UDUT302-4

**Pull request merged**